local_example_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package optimize_test
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/gonum/optimize"
+	"github.com/gonum/optimize/functions"
+)
+
+func ExampleLocal_BFGS() {
+	p := optimize.Problem{
+		Func: functions.ExtendedRosenbrock{}.Func,
+		Grad: functions.ExtendedRosenbrock{}.Grad,
+	}
+
+	method := &optimize.BFGS{}
```

Comment:
I'd move this either in the call to local or just before the call to local, that way it's a bit more obvious.